modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -270,18 +270,21 @@ const removeCaretFormat = (editor: Editor, name: string, vars?: FormatVars, simi
   }
 
   const parents: Node[] = [];
+  let matchCount = 0;
   let formatNode: Element | undefined;
-  while (node) {
+  while (node && isFormatElement(editor, SugarElement.fromDom(node))) {
     if (MatchFormat.matchNode(editor, node, name, vars, similar)) {
-      formatNode = node as Element;
-      break;
-    }
-
-    if (node.nextSibling) {
-      hasContentAfter = true;
+      if (matchCount === 0) {
```

Comment:
Since this is the only place it's checked, it should be a boolean not a number 🤨

Reply:
oh... yeah I thought I changed it. Fixing it now